src/test/ui/consts/offset.rs

```diff
@@ -0,0 +1,88 @@
+// run-pass
+#![feature(const_ptr_offset)]
+use std::ptr::{null, NonNull};
+
+#[repr(C)]
+struct Struct {
+    a: u32,
+    b: u32,
+    c: u32,
+}
+
+const S: Struct = Struct { a: 0, b: 0, c: 0 };
+const fn a() -> *const u32 {
+    &S.a
+}
+const fn c() -> *const u32 {
+    &S.c
+}
+
+// Helper to avoid needing #[feature(const_compare_raw_pointers)]
```

Comment:
Why not use that feature? I think tests should be as readable as possible.

But how would you use it anyway? The final value would still be compared in run-time-code, I presume?

Also, please explain how this type is used. I had to reverse engineer that from the code.

Reply:
I was trying to get this to work with `#[feature(const_compare_raw_pointers)]` but there seem to be some bugs with that feature. I keep getting a `"pointer arithmetic or comparison" needs an rfc` error (even when the feature is enabled).

My use case is just putting the pointers in static data, so I might just try to make the test cases better reflect my use-case.

Reply:
I got this working with `offset_of`, so now the test constants are just bools, which are then checked in the `main` function. I would have preferred using `assert_eq`, but that isn't supported in constants yet.